repowiz-infrastructure/src/main/groovy/life/qbic/repowiz/cli/SubmissionController.groovy

```diff
@@ -1,36 +1,132 @@
 package life.qbic.repowiz.cli
 
-
+import ch.ethz.sis.openbis.generic.asapi.v3.IApplicationServerApi
+import ch.ethz.sis.openbis.generic.dssapi.v3.IDataStoreServerApi
+import life.qbic.repowiz.RepositoryDatabaseConnector
+import life.qbic.repowiz.RepositoryDescription
+import life.qbic.repowiz.SubmissionHandler
+import life.qbic.repowiz.finalise.FinaliseSubmissionImpl
+import life.qbic.repowiz.finalise.SubmissionOutput
+import life.qbic.repowiz.finalise.geo.GeoTemplateParser
+import life.qbic.repowiz.finalise.mapping.RepositoryPluginHandler
+import life.qbic.repowiz.finalise.mapping.SubmissionToPlugin
+import life.qbic.repowiz.finalise.parsing.RepositoryInput
+import life.qbic.repowiz.finalise.parsing.RepositoryOutput
+import life.qbic.repowiz.find.FindMatchingRepositories
 import life.qbic.repowiz.find.FindMatchingRepositoriesInput
+import life.qbic.repowiz.io.JsonParser
+import life.qbic.repowiz.prepare.PrepareSubmissionImpl
+import life.qbic.repowiz.prepare.PrepareSubmissionInput
+import life.qbic.repowiz.prepare.PrepareSubmissionOutput
+import life.qbic.repowiz.prepare.openBis.OpenBisSession
+import life.qbic.repowiz.prepare.openBis.ProjectSearcher
+import life.qbic.repowiz.select.SelectRepository
 import life.qbic.repowiz.select.SelectRepositoryInput
-import life.qbic.repowiz.utils.IO
+import life.qbic.repowiz.finalise.FinaliseSubmission
 
 class SubmissionController {
 
-    String propertiesFile
-    FindMatchingRepositoriesInput findRepoInput
-    SelectRepositoryInput selectRepositoryInput
+    String projectID
+    ProjectSearcher projectSearch
+    RepositoryDescription repoDescription
+    RepositoryPluginHandler repositoryOutput
+
+    SubmissionPresenter presenter
+
+    //use cases
+    FindMatchingRepositoriesInput findRepository
+    SelectRepositoryInput selectRepository
+    PrepareSubmissionInput prepareSubmission
+    FinaliseSubmission finaliseSubmission
+
+    SubmissionController(CommandlineView view, String projectID, String config) {
```

Comment:
what does the config object contain?

Reply:
information about the connection to the local database